### ISSUE TITLE
Bug introduced in 2202.04.22 -- custom __getattr__ fails to find attr shaped_metadata

### ISSUE THREAD

**Issue by JacksonMaxfield** [cgohlke/tifffile#127]
Hey @cgohlke :wave: 

With the recent 2202.04.22, it seems like the breaking changes to `is_shaped` / `shaped_metadata` / `shaped_description` are _really_ breaking. Our CI picked it up: https://github.com/AllenCellModeling/aicsimageio/runs/6165608134?check_suite_focus=true

But on debugging a bit more and adding print statements all over `tifffile` it seems to be within the `getattr`: 
```python
aicsimageio/readers/tiff_glob_reader.py:248: in __init__
    print(tiff.shaped_metadata)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tifffile.TiffFile 'S0_T0_C0_Z0.tif'>, name = 'shaped_metadata'

    def __getattr__(self, name: str, /) -> bool:
        """Return 'is_flag' attributes from first page."""
        print("trying to get", name)
        if name[3:] in TIFF.FILE_FLAGS:
            print("within name thing")
            if not self.pages:
                print("returning fast")
                return False
            print("haven't returned yet")
            value = bool(getattr(self.pages.first, name))
            setattr(self, name, value)
            return value
>       raise AttributeError(f"{self.__class__.__name__!r} object has no attribute {name!r}")
E       AttributeError: 'TiffFile' object has no attribute 'shaped_metadata'
```

I only get back out the following prints:

```
trying to get shaped_description
trying to get shaped_metadata
```

which means it never went into the if statement and just went straight to the AttributeError.

Happy to make a PR if you can tell me the easy fix but I don't know what to change to make the custom __getattr__ to work :sweat_smile: 

Or if I am entirely misreading the code and this wasn't a bug, can you help me figure out how to migrate code from the prior patch to the new patch with the breaking is_shaped / shaped_metadata changes?

Thanks again!

**Comment by cgohlke** [comment]
See also https://github.com/theislab/squidpy/issues/526. It's a regression that escaped unit testing. Will be fixed in the next version (probably tomorrow).

**Comment by JacksonMaxfield** [comment]
Ahhh perfect! Thank you!

**Comment by cgohlke** [comment]
Should be fixed in v2022.4.26.

Version 2022.4.22 added a [smoketest for aicsimageio](https://github.com/cgohlke/tifffile/blob/c97cd9c35494ce7ab941feac9fde055c47740f39/tests/test_tifffile.py#L14845-L14872) but that doesn't touch `TiffGlobReader`. Does `TiffGlobReader` have to rely on `shaped_metadata`?  At https://github.com/AllenCellModeling/aicsimageio/blob/3c53ebc0af152f3a05476852531dbdbfb8231204/aicsimageio/readers/tiff_glob_reader.py#L245-L251, `tiff.shaped_metadata[0]["shape"]` is the same as the more generic `tiff.series[0].shape` used in the other case.
